x-pack/plugins/security_solution/public/app/app.tsx

```diff
@@ -115,13 +117,15 @@ const StartAppComponent: FC<StartAppComponent> = ({
                             <CellActionsProvider
                               getTriggerCompatibleActions={uiActions.getTriggerCompatibleActions}
                             >
-                              <PageRouter
-                                history={history}
-                                onAppLeave={onAppLeave}
-                                setHeaderActionMenu={setHeaderActionMenu}
-                              >
-                                {children}
-                              </PageRouter>
+                              <UpsellingProvider upsellingService={upselling}>
```

Comment:
Is the aim of this to validate if we should show upsell or not on the page level? 
If yes, I think there is already a [Wrapper](https://github.com/elastic/kibana/blob/88aa68aec8847e6eb7e08a6eeb87ed904243e8c0/x-pack/plugins/security_solution/public/common/components/security_route_page_wrapper/index.tsx#L37) for this. Maybe it would suite your needs? :)